tests/phpunit/CRM/Pledge/BAO/PledgeTest.php

```diff
@@ -129,4 +129,59 @@ public function testRetrieveKnownPledgeID() {
     $this->assertEquals(count($pledgeId), 1, "Pledge was retrieved");
   }
 
+  /**
+   *  Test create recur record.
+   */
+  public function testCreateRecurRecord() {
+    //when user creating pledge record.
+    $contributionPageId = ContributionPage::create();
+    $processorId = $this->paymentProcessorAuthorizeNetCreate(array('is_test' => 0));
+    $contribution = $this->contributionCreate(array('contact_id' => $this->_contactId));
+    $pledgeParams = array(
+      'contact_id' => $this->_contactId,
+      'installment_amount' => 100,
+      'actual_amount' => 100,
+      'contribution_id' => $contribution,
+      'contribution_page_id' => $contributionPageId,
+      'financial_type_id' => 1,
+      'frequency_interval' => 1,
+      'installments' => 3,
+      'frequency_unit' => 'month',
+      'frequency_day' => intval(date("d")),
+      'start_date' => date("Ymd"),
+      'create_date' => date("Ymd"),
+      'scheduled_date' => date("Ymd"),
+      'status_id' => 1,
+      'max_reminders' => 1,
+      'initial_reminder_day' => 1,
+      'additional_reminder_day' => 1,
+      'original_installment_amount' => 100,
+    );
+
+    $pledge = CRM_Pledge_BAO_Pledge::create($pledgeParams);
```

Comment:
Hi @eileenmcnaughton,

Cool, I will do that for the unit tests in favour of functions I write in the future. For this though, I really need the $pledge as an object and not an array since the existing function present in https://github.com/JMAConsulting/civicrm-core/blob/MISC-110/CRM/Contribute/Form/Contribution/Confirm.php#L951 uses BAO rather than API. Current budget of the project is not sufficient to rewrite this functionality and do a round of QA.